server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsNodes.java

```diff
@@ -854,4 +858,48 @@ public XContentBuilder toXContent(final XContentBuilder builder, final Params pa
 
     }
 
+    static class PeerRecoveryStats implements ToXContentFragment {
+
+        private long totalStartedRecoveries;
+        private long totalFailedRecoveries;
+        private long totalCompletedRecoveries;
+        private long totalRetriedRecoveries;
+        private long totalCancelledRecoveries;
+
+        PeerRecoveryStats(List<NodeStats> nodeStats) {
+            for (NodeStats nodeStat : nodeStats) {
+                if (nodeStat.getPeerRecoveryStats() != null) {
+                    totalStartedRecoveries += nodeStat.getPeerRecoveryStats().getTotalStartedRecoveries();
+                    totalFailedRecoveries += nodeStat.getPeerRecoveryStats().getTotalFailedRecoveries();
+                    totalCompletedRecoveries += nodeStat.getPeerRecoveryStats().getTotalCompletedRecoveries();
+                    totalRetriedRecoveries += nodeStat.getPeerRecoveryStats().getTotalRetriedRecoveries();
+                    totalCancelledRecoveries += nodeStat.getPeerRecoveryStats().getTotalCancelledRecoveries();
```

Comment:
We can call `nodeStat.getPeerRecoveryStats()` just once and store the result

```suggestion
                    PeerRecoveryStats peerRecoveryStats = nodeStat.getPeerRecoveryStats();
                    totalStartedRecoveries += peerRecoveryStats.getTotalStartedRecoveries();
                    totalFailedRecoveries += peerRecoveryStats.getTotalFailedRecoveries();
                    totalCompletedRecoveries += peerRecoveryStats.getTotalCompletedRecoveries();
                    totalRetriedRecoveries += peerRecoveryStats.getTotalRetriedRecoveries();
                    totalCancelledRecoveries += peerRecoveryStats.getTotalCancelledRecoveries();
```